embedded/sql/catalog.go

```diff
@@ -32,9 +32,16 @@ import (
 type Catalog struct {
 	prefix []byte
 
-	tables       []*Table
-	tablesByID   map[uint32]*Table
-	tablesByName map[string]*Table
+	tables        []*Table
+	tablesByID    map[uint32]*Table
+	tablesByName  map[string]*Table
+	deletedTables map[string]*Table
```

Comment:
this should fail, needs work to be done

Reply:
the id for new table is calculated on length of tables array, which is problematic